src/gas.rs

```diff
@@ -15,7 +15,7 @@ pub enum GasError {
     GasLimitExceeded,
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Copy)]
```

Comment:
Even though it is quite a small data structure - 128bit - I'm not sure I'm a fan of making gas meter `Copy`. Seems prone to error in this kind of logic.

Reply:
Elimitaed copy, so this concern is resolved